src/frameworks/react-i18next.ts

```diff
@@ -22,18 +29,125 @@ class ReactI18nextFramework extends Framework {
 
   // for visualize the regex, you can use https://regexper.com/
   usageMatchRegex = [
-    // general jsx attrs
-    '[^\\w\\d](?:i18nKey=|FormattedMessage[ (]\\s*id=|t\\(\\s*)[\'"`]({key})[\'"`]',
-    '<Trans>({key})<\\/Trans>',
+    '\\Wt\\(\\s*[\'"`]({key})[\'"`]',
+    '\\Wi18nKey=[\'"`]({key})[\'"`]',
+  ]
+
+  derivedKeyRules = [
+    '{key}_plural',
+    '{key}_0',
+    '{key}_1',
+    '{key}_2',
+    '{key}_3',
+    '{key}_4',
+    '{key}_5',
+    '{key}_6',
+    '{key}_7',
+    '{key}_8',
+    '{key}_9',
+    // support v4 format as well as v3
+    '{key}_zero',
+    '{key}_one',
+    '{key}_two',
+    '{key}_few',
+    '{key}_many',
+    '{key}_other'
   ]
 
   refactorTemplates(keypath: string) {
     return [
       `{t('${keypath}')}`,
       `t('${keypath}')`,
+      `<Trans i18nKey="${keypath}"></Trans>`,
       keypath,
     ]
   }
+
+  rewriteKeys(key: string, source: RewriteKeySource, context: RewriteKeyContext = {}) {
+    const dotedKey = key.split(this.namespaceDelimitersRegex).join('.')
```

Comment:
AFAICT, during `i18n-ally.extract-text`, `key`s with either `:` or `/` will fail to get here, since they will be caught by [this call to `keypathValidate`](https://github.com/lokalise/i18n-ally/blob/1d0a020a30a49c3ac7e30ad36bb74eb9302aa49a/src/commands/extractString.ts#L135-L136).

This isn't necessarily a problem for this PR, but something that'll need to be fixed in order to get things working 100% with react-i18n.